twisted/internet/defer.py

```diff
@@ -661,6 +661,66 @@ def __str__(self):
     __repr__ = __str__
 
 
+    def __await__(self):
+        awaiter = _MakeDeferredAwaitable(self)
+        return awaiter.__await__()
+
+
+
+class _MakeDeferredAwaitable(object):
+    """
+    A subgenerator which is created by L{Deferred.__await__}.
+    """
+    result = _NO_RESULT
+
+    def __init__(self, d):
+        self.d = d
+        self.d.addBoth(self._setResult)
+
+    def _setResult(self, value):
+        self.result = value
+
+    def __next__(self):
+        if self.result is not _NO_RESULT:
+            raise StopIteration(self.result)
+        return self.d
+
+    def __await__(self):
```

Comment:
@1st1 right, hmm, that's fixable.

re throw: it's not, because the Deferred will fire with a `Failure`; which `inlineCallbacks` will then throw into it (https://github.com/twisted/twisted/pull/453/files/5ffe7c0aed2c3dcea26ab6c6af8cb1ac7bf5f216#diff-41cf94828242f95ae48c543d7db3309aR1183).